docs/sources/mimir/configure/migrate-ha-tracker-to-memberlist.md

```diff
@@ -0,0 +1,218 @@
+```
+
+**Verification:**
+
+Monitor the following metrics to verify mirroring is working:
+
+- `cortex_multikv_mirror_enabled` – Shows which distributors have mirroring enabled (should be 1 for all)
+- `rate(cortex_multikv_mirror_writes_total[$__rate_interval])` – Rate of writes to secondary store (memberlist)
+- `rate(cortex_multikv_mirror_write_errors_total[$__rate_interval])` – Rate of write errors (should be 0 or very low)
+
+After mirroring is enabled, you should see HA tracker keys in the memberlist cluster information on the `/memberlist` admin page.
+
+### Step 3: Switch to memberlist as primary
+
+Switch the primary and secondary stores so memberlist becomes the primary store.
+
+Update the configuration:
+
+```yaml
+distributor:
+  ha_tracker:
+    enable_ha_tracker: true
+    kvstore:
+      store: multi
+      multi:
+        primary: memberlist # Changed in this step
+        secondary: consul # or etcd - changed in this step
+        mirror_enabled: true
+```
+
+From this point on, distributors will read from memberlist and mirror updates to Consul/etcd.
+
+**Verification:**
+
+Monitor `cortex_multikv_primary_store` to verify all distributors are using memberlist as the primary store. The `store` label should have the value `memberlist`.
+
+### Step 4: Disable mirroring
+
+Stop mirroring writes to Consul/etcd.
+
+Update the configuration:
+
+```yaml
+distributor:
+  ha_tracker:
+    enable_ha_tracker: true
+    kvstore:
+      store: multi
+      multi:
+        primary: memberlist
+        secondary: consul # or etcd
+        mirror_enabled: false # Changed in this step
+```
+
+**Verification:**
+
+Monitor `cortex_multikv_mirror_enabled` to verify all distributors have disabled mirroring (value should be 0).
+
+### Step 5: Remove multi KV store configuration
+
+Configure the HA tracker to use memberlist directly, removing the multi KV store wrapper.
+
+Update the configuration:
+
+```yaml
+distributor:
+  ha_tracker:
+    enable_ha_tracker: true
+    kvstore:
+      store: memberlist # Changed in this step
+      memberlist:
+        join_members:
+          - mimir-gossip-ring.default.svc.cluster.local:7946
+```
+
+Apply this configuration change. This step requires a rollout of all distributor instances.
+
+After the rollout completes, distributors will use memberlist directly for the HA tracker. The `cortex_multikv_*` metrics will no longer be exposed.
+
+### Step 6: Cleanup
+
+You have successfully migrated the HA tracker from Consul/etcd to memberlist without downtime.
+
+If you are no longer using Consul or etcd for any other purpose in your Mimir deployment, you can now:
+
+- Remove the Consul or etcd deployment
+- Remove any Consul/etcd monitoring dashboards and alerts
+- Remove Consul/etcd client configuration from your Mimir configuration
+
+## Helm-specific guidance
+
+If you're using the `mimir-distributed` Helm chart, the migration steps are the same but use Helm values instead of direct YAML configuration:
```

Comment:
```suggestion
If you're using the `mimir-distributed` Helm chart, the migration steps are the same but use Helm values instead of direct YAML configuration. For example:
```